lib/rules/no-warning-comments.js

```diff
@@ -141,12 +141,18 @@ module.exports = {
 
             const matches = commentContainsWarningTerm(node.value);
 
-            matches.forEach(matchedTerm => {
+            matches.forEach(({ type, comment }) => {
+                let commentToDisplay = comment.trim();
+
+                if (commentToDisplay.length >= 30) {
+                    commentToDisplay = `${commentToDisplay.slice(0, 30)}..`;
```

Comment:
Maybe `...` instead of `..`?

Though, I guess this is language-specific punctuation, so it might be technically correct to create a new `messageId` for this case.

Reply:
there is an `.` at the end of the message. so if I am adding `...` it would be `....` then. 
Thats why.

> so it might be technically correct to create a new `messageId` for this case.

actually I am using the existing one as I dont think a new one is needed. if we are putting this update an option, I guess we can then make separate message ids.

Reply:
Ah, it seems that the default formatter doesn't show the last `.` in messages. We should maybe put quotes around the comment (and `...` in quotes) to avoid this.

> > so it might be technically correct to create a new `messageId` for this case.
> 
> actually I am using the existing one as I dont think a new one is needed. if we are putting this update an option, I guess we can then make separate message ids.

It might be nice for translation purposes if `...` isn't common for all languages (I'm not sure about this), though I guess we could improve that later if needed.

Reply:
> Ah, it seems that the default formatter doesn't show the last `.` in messages. We should maybe put quotes around the comment (and `...` in quotes) to avoid this.

Like this 
```
542:5  error  Unexpected 'todo' comment: "* TODO: @aladdin-add...".
```

Reply:
> It might be nice for translation purposes if `...` isn't common for all languages (I'm not sure about this), though I guess we could improve that later if needed.

like js, ts it might be common (not sure though).
any other lang we need to consider ? may be they are out of scope for eslint

Reply:
> > It might be nice for translation purposes if `...` isn't common for all languages (I'm not sure about this), though I guess we could improve that later if needed.
> 
> like js, ts it might be common (not sure though).
> any other lang we need to consider ? may be they are out of scope for eslint

I meant languages other than English, those that have some other punctuation that indicates an intentional omission, but we can audit all the rules and fix things like this later when we precisely define the infrastructure for localization of error messages.

Reply:
> Like this ?
> 
> ```
> 542:5  error  Unexpected 'todo' comment: "* TODO: @aladdin-add...".
> ```

Yes, like that, just with single quotes (I think we're aways using single quotes in error messages).

Reply:
> I meant languages other than English,

ops,  sorry 😄 

> those that have some other punctuation that indicates an intentional omission, but we can audit all the rules and fix things like this later when we precisely define the infrastructure for localization of error messages.

yeah, cool

> Yes, like that, just with single quotes (I think we're aways using single quotes in error messages).

cool

Reply:
done